version: "3"

services:

  nginx-proxy:
    image: jwilder/nginx-proxy
#    restart: on-failure
    container_name: "nginx_proxy_container"
    volumes:
      - nginx-proxy-data:/etc/nginx/vhost.d
      - nginx-proxy-data:/usr/share/nginx/html
      - /var/run/docker.sock:/tmp/docker.sock:ro
      - ./certs:/etc/nginx/certs:ro
    ports:
      - "80:80"
      - "443:443"
    labels:
      - "com.github.jrcs.letsencrypt_nginx_proxy_companion.nginx_proxy"
    privileged: true
    networks:
      - front_bridge

  letsencrypt-nginx:
    image: jrcs/letsencrypt-nginx-proxy-companion
#    restart: on-failure
    container_name: "letsencrypt_nginx_container"
    volumes:
      - nginx-proxy-data:/etc/nginx/vhost.d
      - nginx-proxy-data:/usr/share/nginx/html
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./certs:/etc/nginx/certs:rw
    depends_on:
      - "nginx-proxy"
    networks:
      - front_bridge

networks:
  front_bridge:
    external: true

volumes:
  nginx-proxy-data:
